UnitTests/GitUITests/Theming/ThemeManagerTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using GitExtUtils.GitUI.Theming;
+using GitUI.Theming;
+using NUnit.Framework;
+
+namespace GitUITests.Theming
+{
+    [TestFixture]
+    public class ThemeManagerTests
+    {
+        private static readonly Color InitialColor = Color.Aqua;
+        private static readonly Color CurrentThemeColor = Color.Beige;
+        private static readonly Color ModifiedColor = Color.Crimson;
+
+        private const AppColor AppColor = GitExtUtils.GitUI.Theming.AppColor.Branch;
+        private const KnownColor SysColor = KnownColor.Control;
+
+        private static readonly DefaultTheme DefaultTheme = new DefaultTheme();
+
+        [Test]
+        public void Uses_initial_theme_by_default()
+        {
+            var themeManager = new ThemeManager(DefaultTheme);
+
+            Assert.That(themeManager.UseInitialTheme, Is.True);
+
+            themeManager.SetInitialTheme(CreateTheme(InitialColor));
+
+            Assert.That(themeManager.UseInitialTheme, Is.True);
+            Assert.That(themeManager.IsCurrentThemeInitial(), Is.True);
+        }
+
+        [TestCase(ThemeEditVariant.SelectAnotherTheme)]
+        [TestCase(ThemeEditVariant.ModifyCurrentTheme)]
+        public void When_using_initial_theme_returns_color_from_initial_theme(ThemeEditVariant editVariant)
+        {
+            var themeManager = new ThemeManager(DefaultTheme);
+            themeManager.SetInitialTheme(CreateTheme(InitialColor));
+
+            switch (editVariant)
+            {
+                case ThemeEditVariant.SelectAnotherTheme:
+                    themeManager.SetTheme(CreateTheme(ModifiedColor));
+                    Assert.That(themeManager.IsCurrentThemeInitial(), Is.False);
+                    Assert.That(themeManager.IsCurrentThemeModified(), Is.False);
+                    break;
+                case ThemeEditVariant.ModifyCurrentTheme:
+                    themeManager.SetColor(AppColor, ModifiedColor);
+                    themeManager.SetColor(SysColor, ModifiedColor);
+                    Assert.That(themeManager.IsCurrentThemeInitial(), Is.True);
+                    Assert.That(themeManager.IsCurrentThemeModified(), Is.True);
+                    break;
+                default:
+                    throw new NotSupportedException();
+            }
+
+            int repeats = 3;
```

Comment:
Why 3 repeats?

Reply:
I feel paranoid about Theme manager having or gaining in the future a sort of 'memory', so I decided to have a property switched several (for example 3) times to make sure it's not 'sticky'.

Reply:
added comment